pkgs/applications/networking/cluster/terraform/default.nix

```diff
@@ -28,45 +29,58 @@ let
 
       ldflags = [ "-s" "-w" ];
 
+      subPackages = [ "." ];
+
       postConfigure = ''
         # speakeasy hardcodes /bin/stty https://github.com/bgentry/speakeasy/issues/22
         substituteInPlace vendor/github.com/bgentry/speakeasy/speakeasy_unix.go \
           --replace "/bin/stty" "${coreutils}/bin/stty"
       '';
 
+      nativeBuildInputs = [ installShellFiles ];
+
+      preCheck = ''
+        export HOME=$TMPDIR
+        export TF_SKIP_REMOTE_TESTS=1
+      '';
+
+      # https://github.com/posener/complete/blob/9a4745ac49b29530e07dc2581745a218b646b7a3/cmd/install/bash.go#L8
+      completionBash = "complete -C terraform terraform\n";
+      # https://github.com/posener/complete/blob/9a4745ac49b29530e07dc2581745a218b646b7a3/cmd/install/zsh.go
+      completionZsh =  "complete -C terraform terraform\n";
+      # https://github.com/posener/complete/blob/9a4745ac49b29530e07dc2581745a218b646b7a3/cmd/install/fish.go#L56
+      completionFish = ''
+        function __complete_terraform
+            set -lx COMP_LINE (commandline -cp)
+            test -z (commandline -ct)
+            and set COMP_LINE "$COMP_LINE "
+            terraform
+        end
+        complete -f -c terraform -a "(__complete_terraform)"
+      '';
+
+      passAsFile = [ "completionBash" "completionZsh" "completionFish" ];
+
       postInstall = ''
         # remove all plugins, they are part of the main binary now
         for i in $out/bin/*; do
           if [[ $(basename $i) != terraform ]]; then
             rm "$i"
           fi
         done
-      '';
 
-      preCheck = ''
-        export HOME=$TMPDIR
-        export TF_SKIP_REMOTE_TESTS=1
+        installShellCompletion --bash --name terraform $completionBashPath
+        installShellCompletion --zsh --name terraform $completionZshPath
+        installShellCompletion --fish --name terraform $completionFishPath
       '';
 
-      subPackages = [ "." ];
-
-      meta = with lib; {
+      meta = {
         description =
           "Tool for building, changing, and versioning infrastructure";
         homepage = "https://www.terraform.io/";
         changelog = "https://github.com/hashicorp/terraform/blob/v${version}/CHANGELOG.md";
-        license = licenses.mpl20;
-        maintainers = with maintainers; [
```

Comment:
Why was this reformatted?

Reply:
To remove double `with` at meta.
Reasoning: https://github.com/atorres1985-contrib/nixpkgs/commit/451dfa6d84586b5cb83e6a7254689dff66b4523a

Reply:
That isn't nixpkgs policy.

Reply:
> That isn't nixpkgs policy.

And what is the policy?

Reply:
Feel free to revert whatever you want.

Reply:
That's to remove the `with`. That was @superherointj idea, though.